module/zfs/dbuf.c

```diff
@@ -3530,15 +3530,17 @@ dbuf_write_children_ready(zio_t *zio, arc_buf_t *buf, void *vdb)
 	dnode_t *dn;
 	blkptr_t *bp;
 	uint64_t i;
+	uint64_t blknum;
 	int epbs;
 
 	ASSERT3U(db->db_level, >, 0);
 	DB_DNODE_ENTER(db);
 	dn = DB_DNODE(db);
 	epbs = dn->dn_phys->dn_indblkshift - SPA_BLKPTRSHIFT;
+	blknum = 1ULL << epbs;
 
 	/* Determine if all our children are holes */
-	for (i = 0, bp = db->db.db_data; i < 1 << epbs; i++, bp++) {
+	for (i = 0, bp = db->db.db_data; i < blknum; i++, bp++) {
```

Comment:
Actually, I was just suggesting adding the ULL suffix to the existing code.  The local variable works but doing it inline minimized the difference between us and upstream.  For example:

```diff
-	for (i = 0, bp = db->db.db_data; i < 1 << epbs; i++, bp++) {
+	for (i = 0, bp = db->db.db_data; i < 1ULL << epbs; i++, bp++) {
￼```